components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -69,20 +105,39 @@ void MediaDetectorComponentManager::RegisterIfNeeded() {
 
 void MediaDetectorComponentManager::OnComponentReady(
     const base::FilePath& install_path) {
+  base::flat_set<base::FilePath> files(
+      {install_path.Append(GetBaseScriptName())});
+  for (const auto& [file, _] : GetScriptNameToSchemefulSiteMap()) {
+    files.insert(install_path.Append(file));
+  }
+
   base::ThreadPool::PostTaskAndReplyWithResult(
       FROM_HERE, base::MayBlock(),
-      base::BindOnce(&ReadScript, GetScriptPath(install_path)),
-      base::BindOnce(&MediaDetectorComponentManager::OnGetScript,
+      base::BindOnce(&ReadScriptsFromComponent, files),
+      base::BindOnce(&MediaDetectorComponentManager::OnGetScripts,
                      weak_factory_.GetWeakPtr()));
 }
 
-void MediaDetectorComponentManager::OnGetScript(const std::string& script) {
-  if (script.empty()) {
-    LOG(ERROR) << __FUNCTION__ << " script is empty!";
+void MediaDetectorComponentManager::OnGetScripts(
+    const MediaDetectorComponentManager::ScriptMap& script_map) {
+  if (script_map.empty()) {
+    LOG(ERROR) << __FUNCTION__ << " scripts are empty!";
     return;
   }
 
-  script_ = script;
+  DCHECK(script_map.count(GetBaseScriptName()));
+  script_ = script_map.at(GetBaseScriptName());
```

Comment:
nit: As we'll have base and site specific scripts, how about `base_script_`?

Reply:
That's a way better name. Changed.